apps/docs/scripts/typedoc-postbuild.ts

```diff
@@ -1,17 +1,140 @@
+              items: [],
+            });
+          }
+        });
+    });
+
+  writeFileSync('.typedoc/api-links.json', JSON.stringify(links));
+};
+
+/**
+ * Alters the typedoc generated file structure to be more semantic.
+ */
+const alterFileStructure = () => {
+  const modulesFiles = readdirSync(modulesDir);
+  const classesFiles = readdirSync(classesDir);
+
+  modulesFiles.forEach((modulesFile) => {
+    // Create a new directory for each module
+    const newDirName = modulesFile.split('.')[0];
+    const newDirPath = join(apiDocsDir, newDirName);
+    mkdirSync(newDirPath);
+
+    // Copy the class files to the correct module directory
+    classesFiles.forEach((classesFile) => {
+      if (classesFile.startsWith(newDirName)) {
+        const newDirClassFilePath = join(newDirPath, classesFile);
+        copyFileSync(join(classesDir, classesFile), newDirClassFilePath);
+
+        // Rename the class file to remove module prefix
+        renameSync(newDirClassFilePath, join(newDirPath, classesFile.split('-')[1]));
+      }
+    });
+
+    // Move module index file
+    copyFileSync(join(modulesDir, modulesFile), join(newDirPath, 'index.md'));
+
+    // Rename module directory to remove 'fuel_ts_' prefix
+    const formattedDirName = newDirPath.split('fuel_ts_')[1];
+    const capitalisedDirName = formattedDirName.charAt(0).toUpperCase() + formattedDirName.slice(1);
+    renameSync(newDirPath, join(apiDocsDir, capitalisedDirName));
   });
-});
+};
+
+/**
+ * Recreates the generated typedoc links
+ * 
+ * TODO: this should be done via the filesystem rather than hardcoded
+ */
+const recreateInternalLinks = () => {
+  const regexReplaces = [
+    { regex: '../modules/interfaces.md', replacement: './index.md' },
+    { regex: '../modules/fuel_ts_address.md', replacement: './index.md' },
+
+    { regex: '../classes/fuel_ts_address-Address.md', replacement: './Address.md' },
+    { regex: 'fuel_ts_address-Address.md', replacement: './Address.md' },
+
+    { regex: '../classes/fuel_ts_interfaces-AbstractAccount.md', replacement: './AbstractAccount.md' },
+    { regex: 'interfaces-AbstractAccount.md', replacement: './AbstractAccount.md' },
+
+    { regex: '../classes/fuel_ts_interfaces-AbstractAddress.md', replacement: './AbstractAddress.md' },
+    { regex: 'interfaces-AbstractAddress.md', replacement: 'AbstractAddress.md' },
+
+    { regex: '../classes/fuel_ts_interfaces-AbstractContract.md', replacement: './AbstractContract.md' },
+    { regex: 'interfaces-AbstractContract.md', replacement: './AbstractContract.md' },
+  ];
+
+  const topLevelDirs = readdirSync(apiDocsDir);
+
+  topLevelDirs
+    .filter((dir) => dir !== 'index.md')
+    .forEach((dir) => {
+      regexReplaces.forEach(({ regex, replacement }) => {
+        replace({
+          regex,
+          replacement,
+          paths: [join(apiDocsDir, dir)],
+          recursive: true,
+          silent: true,
+        });
+      });
+    });
+};
```

Comment:
This whole function is WIP, using hardcoded values to unblock the doc hub project as the main blocker for them is our file structure. I will produce a more automated function this week.